### PR TITLE
Cluster-autoscaler: check if cluster is healthy and add new node lister

### PR DESCRIPTION
**Pull request by mwielgus**
Ref: #2228 #2229

cc: @jszczepkowski @fgrzadkowski @piosz

### CODE REVIEW

File: cluster-autoscaler/cluster_autoscaler.go [hunk 1]
```diff
@@ -243,23 +244,42 @@ func run(_ <-chan struct{}) {
...
+					glog.Errorf("No ready nodes in the cluster")
+					continue
+				}
+
+				allNodes, err := allNodeLister.List()
+				if err != nil {
+					glog.Errorf("Failed to list all nodes: %v", err)
+					continue
+				}
+				if len(readyNodes) == 0 {
```
Comment (jszczepkowski) [inline cluster-autoscaler/cluster_autoscaler.go]: s/readyNode/allNodes/ ?

File: cluster-autoscaler/utils/kubernetes/listers.go [hunk 2]
```diff
@@ -122,3 +122,32 @@ func NewNodeLister(kubeClient client.Interface) *ReadyNodeLister {
...
+	}
+	return allNodes, nil
+}
+
+// NewAllNodeLister builds a node lister that returns all nodes (ready and unready)
+func NewAllNodeLister(kubeClient client.Interface) *AllNodeLister {
+	listWatcher := cache.NewListWatchFromClient(kubeClient.Core().RESTClient(), "nodes", apiv1.NamespaceAll, fields.Everything())
+	nodeLister := &cache.StoreToNodeLister{Store: cache.NewStore(cache.MetaNamespaceKeyFunc)}
+	reflector := cache.NewReflector(listWatcher, &apiv1.Node{}, nodeLister.Store, time.Hour)
+	reflector.Run()
```
Comment (mwielgus) [inline cluster-autoscaler/utils/kubernetes/listers.go]: It starts its own goroutine in Run

File: cluster-autoscaler/cluster_autoscaler.go (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: Done.

**Comment by jszczepkowski** [comment]
LGTM

**Pull request merged**